src/node/mod.rs

```diff
@@ -0,0 +1,382 @@
+    let stat = stat(path.as_ref()).map_err(Into::<io::Error>::into)?;
+    dev_path(stat.st_rdev, ty)
+}
+
+/// Returns the path of a specific type of node from the same DRM device as an existing [`DrmNode`].
+pub fn node_path(node: &DrmNode, ty: NodeType) -> io::Result<PathBuf> {
+    dev_path(node.dev, ty)
+}
+
+/// Returns the path of a specific type of node from the DRM device described by major and minor device numbers.
+#[cfg(target_os = "linux")]
+pub fn dev_path(dev: dev_t, ty: NodeType) -> io::Result<PathBuf> {
+    use std::fs;
+    use std::io::ErrorKind;
+
+    if !is_device_drm(dev) {
+        return Err(io::Error::new(
+            ErrorKind::NotFound,
+            format!("{}:{} is no DRM device", major(dev), minor(dev)),
+        ));
+    }
+
+    let read = fs::read_dir(format!(
+        "/sys/dev/char/{}:{}/device/drm",
+        major(dev),
+        minor(dev)
+    ))?;
+
+    for entry in read.flatten() {
+        let name = entry.file_name();
+        let name = name.to_string_lossy();
+
+        // Only 1 primary, control and render node may exist simultaneously, so the
+        // first occurrence is good enough.
+        if name.starts_with(ty.minor_name_prefix()) {
+            let path = [r"/", "dev", "dri", &name].iter().collect::<PathBuf>();
+            if path.exists() {
+                return Ok(path);
+            }
+        }
+    }
+
+    Err(io::Error::new(
+        ErrorKind::NotFound,
+        format!(
+            "Could not find node of type {} from DRM device {}:{}",
+            ty,
+            major(dev),
+            minor(dev)
+        ),
+    ))
+}
+
+/// Returns the path of a specific type of node from the DRM device described by major and minor device numbers.
+#[cfg(target_os = "freebsd")]
+fn dev_path(dev: dev_t, ty: NodeType) -> io::Result<PathBuf> {
+    // Based on libdrm `drmGetMinorNameForFD`. Should be updated if the code
+    // there is replaced with anything more sensible...
+
+    use std::io::ErrorKind;
+
+    if !is_device_drm(dev) {
+        return Err(io::Error::new(
+            ErrorKind::NotFound,
+            format!("{}:{} is no DRM device", major(dev), minor(dev)),
+        ));
+    }
+
+    if let Some(dev_name) = devname(dev) {
+        let suffix = dev_name.trim_start_matches(|c: char| !c.is_numeric());
+        if let Ok(old_id) = suffix.parse::<u32>() {
+            let id_mask = 0b11_1111;
+            let id = old_id & id_mask + get_minor_base(ty);
+            let path = PathBuf::from(format!("/dev/dri/{}{}", ty.minor_name_prefix(), id));
+            if path.exists() {
+                return Ok(path);
+            }
+        }
+    }
+
+    Err(io::Error::new(
+        ErrorKind::NotFound,
+        format!(
+            "Could not find node of type {} from DRM device {}:{}",
+            ty,
+            major(dev),
+            minor(dev)
+        ),
+    ))
+}
+
+#[cfg(target_os = "openbsd")]
+fn dev_path(dev: dev_t, ty: NodeType) -> io::Result<PathBuf> {
```

Comment:
For consistency with the others?

```suggestion

/// Returns the path of a specific type of node from the DRM device described by major and minor device numbers.
#[cfg(target_os = "openbsd")]
fn dev_path(dev: dev_t, ty: NodeType) -> io::Result<PathBuf> {
```